bson/bsoncodec/struct_codec.go

```diff
@@ -105,6 +100,11 @@ func NewStructCodec(p StructTagParser, opts ...*bsonoptions.StructCodecOptions)
 	return codec, nil
 }
 
+func newDefaultStructCodec() *StructCodec {
+	codec, _ := NewStructCodec(DefaultStructTagParser)
```

Comment:
We know that `NewStructCodec` only errors if the struct tag parser parameter is nil, so we can safely ignore the error. I didn't think this was too risky because this is an unexported function, so it seems OK to rely on implementation details, but we can also panic if there is an error. Note that we can't return an error because this is all called from builder-style code, but panicking is probably OK too because it's setup code. Anyone have a preference?

Reply:
I would lean towards panicking, but I don't have a strong preference

Reply:
Changed to panic. I also moved it to `default_value_decoders_go` so it's clear that it's used for default codec registration.